apps/api/src/app/environments/usecases/get-my-environments/get-my-environments.usecase.ts

```diff
@@ -11,6 +11,16 @@ export class GetMyEnvironments {
   async execute(command: GetMyEnvironmentsCommand): Promise<EnvironmentEntity[]> {
     Logger.verbose('Getting Environments');
 
-    return await this.environmentRepository.findOrganizationEnvironments(command.organizationId);
+    const environments = await this.environmentRepository.findOrganizationEnvironments(command.organizationId);
+
+    return environments.map((environment) => {
+      if (environment._id === command.environmentId) {
+        return environment;
+      }
```

Comment:
Sorry for raising it now and not before but if we already have the environment id here maybe we could do `environmentRepository.findById` this way we won't need to iterate over a list. (reduce some of the complexity), once again apologies for raising it now and not before 🙏

Reply:
@djabarovgeorge we shouldn't because this endpoint is still supposed to return ALL environments for specific organization, not just one with specific id. This `.map()` just transforms other environments than "current" to hide api keys field.

Reply:
Oh right right, sorry missed this one now. 😅